### PR TITLE
[PL] Use std::function to give the flux calculation to CalculateSurfaceFlux.

### PR DESCRIPTION
**Pull request by TomFischer**
As titled.

Fixes issue #2134.

### CODE REVIEW

File: ProcessLib/HT/HTProcess.cpp [hunk 1]
```diff
@@ -288,7 +288,13 @@ void HTProcess::postTimestepConcreteProcess(GlobalVector const& x,
         getProcessVariables(process_id)[0].get().getNumberOfComponents(),
         _integration_order);
 
-    balance.integrate(x, *balance_pv, t, *this);
+    namespace PH = std::placeholders;
```
Comment (endJunction) [inline ProcessLib/HT/HTProcess.cpp]: ?

File: ProcessLib/CalculateSurfaceFlux/CalculateSurfaceFluxLocalAssembler.h [hunk 2]
```diff
@@ -103,13 +106,20 @@ class CalculateSurfaceFluxLocalAssembler final
...
-                   MeshLib::PropertyVector<double>& balance,
-                   double const t,
-                   Process const& bulk_process) override
+    /// @param bulk_mesh Stores a reference to the bulk mesh that is needed to
+    /// fetch the information for the integration over the surface mesh.
+    /// @param getFlux function that calculates the flux in the integration
+    /// points of the face elements of the bulk element that belongs to the
+    /// surface.
+    void integrate(
+        std::size_t element_id,
```
Comment (endJunction) [inline ProcessLib/CalculateSurfaceFlux/CalculateSurfaceFluxLocalAssembler.h]: const

File: ProcessLib/CalculateSurfaceFlux/CalculateSurfaceFluxLocalAssembler.h [hunk 3]
```diff
@@ -28,11 +28,14 @@ class CalculateSurfaceFluxLocalAssemblerInterface
 public:
     virtual ~CalculateSurfaceFluxLocalAssemblerInterface() = default;
 
-    virtual void integrate(std::size_t element_id,
-                           GlobalVector const& x,
-                           MeshLib::PropertyVector<double>& balance,
-                           double const t,
-                           Process const& bulk_process) = 0;
+    virtual void integrate(
+        std::size_t element_id,
```
Comment (endJunction) [inline ProcessLib/CalculateSurfaceFlux/CalculateSurfaceFluxLocalAssembler.h]: const

File: ProcessLib/CalculateSurfaceFlux/CalculateSurfaceFlux.cpp [hunk 4]
```diff
@@ -53,16 +53,19 @@ CalculateSurfaceFlux::CalculateSurfaceFlux(
...
-                                     MeshLib::PropertyVector<double>& balance,
-                                     double const t,
-                                     Process const& bulk_process)
+void CalculateSurfaceFlux::integrate(
+    GlobalVector const& x,
+    MeshLib::PropertyVector<double>& balance,
+    double const t,
+    MeshLib::Mesh const& bulk_mesh,
+    std::function<Eigen::Vector3d(std::size_t const, MathLib::Point3d const&,
+                                  double const, GlobalVector const&)> getFlux)
```
Comment (chleh) [inline ProcessLib/CalculateSurfaceFlux/CalculateSurfaceFlux.cpp]: Maybe pass `getFlux` by reference?

**Pull request merged**